File: app/src/main/java/org/techtown/find_gas_station/Comparator/OilRoadDistanceComparator.java
package org.techtown.find_gas_station.Comparator;

import org.techtown.find_gas_station.OilList;

import java.util.Comparator;

public class OilRoadDistanceComparator implements Comparator<OilList> {
    @Override
    public int compare(OilList t1, OilList t2) {

        if(Integer.parseInt(t1.getActDistance()) == Integer.parseInt(t2.getActDistance())){
            return Integer.parseInt(t1.getSpendTime()) - Integer.parseInt(t2.getSpendTime());
        }else{
            return Integer.parseInt(t1.getActDistance()) - Integer.parseInt(t2.getActDistance());
        }

    }



}

File: app/src/main/java/org/techtown/find_gas_station/util/comparator/OilRoadDistanceComparator.kt
package org.techtown.find_gas_station.util.comparator

import org.techtown.find_gas_station.domain.stationsInfo.model.TotalOilInfo


class OilRoadDistanceComparator : Comparator<TotalOilInfo> {
    override fun compare(t1 : TotalOilInfo, t2 : TotalOilInfo) =
        if (t1.actDistance.toInt() == t2.actDistance.toInt()) {
            t1.price.toInt() - t2.price.toInt()
        } else {
            t1.actDistance.toInt() - t2.actDistance.toInt()
        }

}